src/cats/CatSlideshow.js

```diff
@@ -0,0 +1,219 @@
+      return initCatSlideshowState();
+    case "update-max-index":
+      return { ...state, maxIndex: action.maxIndex };
+    case "decrement-index":
+      return changeIndex(state, -1);
+    case "increment-index": {
+      return changeIndex(state, 1);
+    }
+    case "prefetch-status": {
+      const newState = {
+        ...state,
+        indexReadyMap: { ...state.indexReadyMap, [action.index]: "ready" },
+      };
+
+      if (action.index === state.index) {
+        newState.visibleIndex = action.index;
+      }
+
+      return newState;
+    }
+    default:
+      throw new Error(`Unhandled action type: ${action.type}`);
+  }
+}
+
+function CatSlideshow({ selectedBreedID }) {
+  const prevSelectedBreedIDRef = useRef(selectedBreedID);
+  const prefetchMapRef = useRef(new Map());
+  const { pages, metadata, fetchPage, resetPages } = usePaginatedFetch(
+    PAGE_SIZE,
+    "loading"
+  );
+
+  const [state, dispatch] = useReducer(
+    catSlideshowReducer,
+    undefined,
+    initCatSlideshowState
+  );
+
+  const { direction, index, visibleIndex } = state;
+
+  useEffect(() => {
+    if (selectedBreedID !== prevSelectedBreedIDRef.current) {
+      prevSelectedBreedIDRef.current = selectedBreedID;
+      resetPages(selectedBreedID);
+      dispatch({ type: "change-breed", selectedBreedID });
+      prefetchMapRef.current.clear();
+    }
+  }, [selectedBreedID, resetPages]);
+
+  useEffect(() => {
+    const pageIndex = getPageIndex(index);
+    const prefetchPageIndex = getPageIndex(index + PAGE_PREFETCH);
+    const doFetch = (i) => fetchPageFromCatsApi(i, selectedBreedID, fetchPage);
+
+    if (!pages[pageIndex]) {
+      doFetch(pageIndex);
+    }
+
+    if (prefetchPageIndex !== pageIndex && !pages[prefetchPageIndex]) {
+      doFetch(prefetchPageIndex);
+    }
+  }, [index, pages, selectedBreedID, fetchPage]);
+
+  useEffect(() => {
+    const prefetchMap = prefetchMapRef.current;
+    const waitIndexes = [...prefetchMap.keys()].filter(
+      (k) => prefetchMap.get(k) === "wait"
+    );
+    const prefetchCount = [...prefetchMap.values()].filter(
+      (v) => v === "prefetch"
+    ).length;
+
+    for (let i = 0; i < MAX_PREFETCH; i++) {
+      const newIndex = index + i + (index === 0 ? 0 : 1);
+      if (!prefetchMap.get(newIndex)) {
+        prefetchMap.set(newIndex, "wait");
+      }
+    }
+
+    if (prefetchCount < MAX_PREFETCH) {
+      waitIndexes.slice(0, MAX_PREFETCH - prefetchCount).forEach((i) => {
+        const url = getImageURL(pages, i);
+
+        const updateImageStatus = (i, status) => {
+          prefetchMap.set(i, status);
+          dispatch({ type: "prefetch-status", index: i, status });
+        };
+
+        if (url) {
+          prefetchMap.set(i, "prefetch");
+
+          preloadImage(url)
+            .then(() => updateImageStatus(i, "ready"))
+            .catch(() => updateImageStatus(i, "error"));
+        }
+      });
+    }
+  }, [pages, index, visibleIndex, state.indexReadyMap]);
```

Comment:
visibleIndex isn't used either

Reply:
I guess the lint plugin is great at finding missing deps but not so great at finding unused deps.

Reply:
Yeah, that is true. Sometimes it is because they can be legitimate (you want the effect to rerun when a dependency is changed, even if it isn't used directly)